src/client/init/root.tsx

```diff
@@ -19,34 +19,53 @@ function useHydrated() {
 
 export const defineClientConfig = (config: RdtClientConfig) => config
 
-export const withDevTools = (Component: any, config?: RemixDevToolsProps) => () => {
-	hydrationDetector()
-	const hydrated = useHydrated()
-	if (!hydrated) return <Component />
-
-	return (
-		<>
-			<Component />
-			{createPortal(<RemixDevTools {...config} />, document.body)}
-		</>
-	)
-}
-
-/**
- *
- * @description Injects the dev tools into the Vite App, ONLY meant to be used by the package plugin, do not use this yourself!
- */
-export const withViteDevTools = (Component: any, config?: RemixDevToolsProps) => () => {
-	hydrationDetector()
-	function AppWithDevTools(props: any) {
+export const withDevTools =
+	(Component: any, config?: RemixDevToolsProps & { panelMode?: "auto" | "embedded" }) => () => {
+		hydrationDetector()
 		const hydrated = useHydrated()
+		const isEmbedded = useIsEmbeddedRDT()
 		if (!hydrated) return <Component />
+
+		if ((config?.panelMode ?? "auto") === "embedded") {
```

Comment:
why the default here? you don't really need to check if "auto" === "embedded"

Reply:
Mainly for clarity that the default mode is `"auto"`. It's not technically necessary, happy to remove it if preferred.